src/Protocol/PortableContact.php

```diff
@@ -1355,11 +1355,12 @@ public static function checkServer($server_url, $network = "", $force = false)
 		$info = strip_tags($info);
 		$platform = strip_tags($platform);
 
-		$fields = ['url' => $server_url, 'version' => $version,
-				'site_name' => $site_name, 'info' => $info, 'register_policy' => $register_policy,
-				'poco' => $poco, 'noscrape' => $noscrape, 'network' => $network,
-				'platform' => $platform, 'registered-users' => $registered_users,
-				'last_contact' => $last_contact, 'last_failure' => $last_failure];
+		$fields = ['url' => $server_url, 'version' => defaults($version, ''),
```

Comment:
I'd rather have earlier variable initialization than this unreadable wall of code.

Reply:
All these variables are initialized before. But some of them are fetched via some JSON object. So they could return ```null``` after that.

Reply:
What I mean is move this block of code outside the condition to initialize the variables to non-NULL values: https://github.com/annando/friendica/blob/8700d2a8ad2412525523acdd970cd1724dab1193/src/Protocol/PortableContact.php#L953-L960

Reply:
This won't help, exactly these values are fetched from the table in the other condition.

Reply:
Then there can't be NULL values for non-NULL columns, right?

Reply:
I changed some things.